File: Script/clean_data.R
# Clean SUoA dataset by removing supporting and reasoning columns
# Keeps only the main variables for analysis

# Source functions
source(here::here("Script", "Functions.R"))

library(openxlsx)
library(dplyr)
library(here)

# File paths - Using CSV to avoid Excel lock issues
input_file <- here("Data", "20260106_working.csv")
output_file <- here("Data", "20260113_cleaned.xlsx")
csv_file <- here("Data", "20260113_cleaned.csv")

# Check if file exists
if (!file.exists(input_file)) {
  stop("ERROR: Input file not found: ", input_file)
}

cat("Reading file:", input_file, "\n\n")

# Read CSV (no lock issues)
df <- read.csv(input_file, stringsAsFactors = FALSE)

cat("\nOriginal dataset shape:", nrow(df), "rows x", ncol(df), "columns\n")
cat("Total columns:", ncol(df), "\n")

# Identify columns to remove (containing 'supporting' or 'reasoning')
columns_to_remove <- grep("supporting|reasoning|Supporting|Reasoning", 
                          names(df), 
                          value = TRUE, 
                          ignore.case = TRUE)

cat("\n", paste(rep("=", 80), collapse = ""), "\n")
cat("COLUMNS TO REMOVE (", length(columns_to_remove), "):\n", sep = "")
cat(paste(rep("=", 80), collapse = ""), "\n")
for (i in seq_along(columns_to_remove)) {
  cat(sprintf("%3d. %s\n", i, columns_to_remove[i]))
}

# Remove the columns
df_cleaned <- df %>% select(-all_of(columns_to_remove))

cat("\n", paste(rep("=", 80), collapse = ""), "\n")
cat("REMAINING COLUMNS (", ncol(df_cleaned), "):\n", sep = "")
cat(paste(rep("=", 80), collapse = ""), "\n")
for (i in seq_along(names(df_cleaned))) {
  cat(sprintf("%3d. %s\n", i, names(df_cleaned)[i]))
}

cat("\n", paste(rep("=", 80), collapse = ""), "\n")
cat("SUMMARY:\n")
cat(paste(rep("=", 80), collapse = ""), "\n")
cat("Original columns: ", ncol(df), "\n", sep = "")
cat("Removed columns:  ", length(columns_to_remove), "\n", sep = "")
cat("Remaining columns:", ncol(df_cleaned), "\n", sep = "")
cat("Rows:            ", nrow(df_cleaned), "\n", sep = "")

# Save cleaned dataset
cat("\nSaving cleaned dataset to:", output_file, "\n")
write.xlsx(df_cleaned, output_file)

# Also save as CSV for easier inspection
cat("Also saving as CSV:", csv_file, "\n")
write.csv(df_cleaned, csv_file, row.names = FALSE)

cat("\n✅ CLEANING COMPLETE!\n")
cat("\nNext steps:\n")
cat("1. Review the cleaned file:", output_file, "\n")
cat("2. Verify all necessary variables are retained\n")
cat("3. Begin analysis with the clean dataset\n")

File: clean_data.py
"""
Clean SUoA dataset by removing supporting and reasoning columns
Keeps only the main variables for analysis
"""

import pandas as pd
import os

# File paths
input_file = r"c:\Users\kukumar\OneDrive - UGent\My Projects\Grafitti Project\Urban_Foraging_Archive\Articles_drafts\2024 Snatching\Data\2026013_working.xlsx"
output_file = r"c:\Users\kukumar\OneDrive - UGent\My Projects\Grafitti Project\Urban_Foraging_Archive\Articles_drafts\2024 Snatching\Data\20260113_cleaned.xlsx"

print(f"Reading file: {input_file}")
df = pd.read_excel(input_file)

print(f"\nOriginal dataset shape: {df.shape}")
print(f"Total columns: {df.shape[1]}")

# Identify columns to remove (containing 'supporting', 'reasoning', 'Supporting', 'Reasoning')
columns_to_remove = [col for col in df.columns if
                     'supporting' in col.lower() or
                     'reasoning' in col.lower() or
                     'Supporting' in col or
                     'Reasoning' in col]

print(f"\n{'='*80}")
print(f"COLUMNS TO REMOVE ({len(columns_to_remove)}):")
print(f"{'='*80}")
for i, col in enumerate(columns_to_remove, 1):
    print(f"{i:3d}. {col}")

# Remove the columns
df_cleaned = df.drop(columns=columns_to_remove)

print(f"\n{'='*80}")
print(f"REMAINING COLUMNS ({len(df_cleaned.columns)}):")
print(f"{'='*80}")
for i, col in enumerate(df_cleaned.columns, 1):
    print(f"{i:3d}. {col}")

print(f"\n{'='*80}")
print(f"SUMMARY:")
print(f"{'='*80}")
print(f"Original columns: {df.shape[1]}")
print(f"Removed columns:  {len(columns_to_remove)}")
print(f"Remaining columns: {df_cleaned.shape[1]}")
print(f"Rows: {df_cleaned.shape[0]}")

# Save cleaned dataset
print(f"\nSaving cleaned dataset to: {output_file}")
df_cleaned.to_excel(output_file, index=False)

# Also save as CSV for easier inspection
csv_file = output_file.replace('.xlsx', '.csv')
print(f"Also saving as CSV: {csv_file}")
df_cleaned.to_csv(csv_file, index=False)

print(f"\n✅ CLEANING COMPLETE!")
print(f"\nNext steps:")
print(f"1. Review the cleaned file: {output_file}")
print(f"2. Verify all necessary variables are retained")
print(f"3. Begin analysis with the clean dataset")
